File: UISamples/Bit6UIFullDemo-ObjC/ChatDemo/DemoContactSource.m
//
//  Bit6DemoContactSource.m
//  Bit6UI
//
//  Created by Carlos Thurber on 12/22/15.
//  Copyright © 2015 Bit6. All rights reserved.
//

#import "DemoContactSource.h"

@implementation DemoContactSource

- (NSDictionary<NSString*,id<BXUContact>>*)source{
    if (!_source) {
        #warning here you can set the displaynames and avatars for each user
        _source = @{
                    @"usr:alex":[[DemoContact alloc] initWithURI:@"usr:alex" name:@"Alexey Goloshubin" avatarURLString:nil],
                    @"usr:julia":[[DemoContact alloc] initWithURI:@"usr:julia" name:@"Julia Goloshubin" avatarURLString:nil],
                    @"usr:calitb":[[DemoContact alloc] initWithURI:@"usr:calitb" name:@"Carlos Thurber" avatarURLString:nil]
                    };
    }
    return _source;
}

- (id<BXUContact>)contactForURI:(NSString *)uri
{
    return self.source[uri];
}

@end

@implementation DemoContact

- (nonnull instancetype)initWithURI:(nonnull NSString*)uri name:(nonnull NSString*)name avatarURLString:(nullable NSString*)avatarURLString
{
    self = [super init];
    if (self) {
        self.uri = uri;
        self.name = name;
        self.avatarURL = [NSURL URLWithString:avatarURLString];
    }
    return self;
}

@end

File: UISamples/Bit6UIFullDemo-Swift/Bit6ChatDemo-Swift/DemoContactSource.swift
//
//  DemoContactSource.swift
//  Bit6UIFullDemo
//
//  Created by Carlos Thurber on 04/20/16.
//  Copyright © 2016 Bit6. All rights reserved.
//

import Foundation
import Bit6UI

class DemoContactSource : NSObject, BXUContactSource {
 
    func contactForURI(uri: String) -> BXUContact? {
        return source[uri]
    }
    
    //WARNING:  here you can set the displaynames and avatars for each user
    lazy var source = ["usr:alex" : DemoContact(uri:"usr:alex", name:"Alexey Goloshubin", avatarURL:nil),
                       "usr:julia": DemoContact(uri:"usr:julia", name:"Julia Goloshubin", avatarURL:nil),
                       "usr:calitb": DemoContact(uri:"usr:calitb", name:"Carlos Thurber", avatarURL:nil)]
    
}

class DemoContact : NSObject, BXUContact {
    
    var _uri: String
    var _name: String
    var _avatarURL : NSURL?
    
    init(uri:String, name:String, avatarURL:NSURL?) {
        _uri = uri
        _name = name
        _avatarURL = avatarURL
    }
    
    func uri() -> String {
        return _uri
    }
    
    func name() -> String {
        return _name
    }
    
    func avatarURL() -> NSURL? {
        return _avatarURL
    }
    
}